tests/template_tests/filter_tests/test_floatformat.py

```diff
@@ -78,6 +74,31 @@ def test_inputs(self):
         self.assertEqual(floatformat(1.5e-15, -20), "0.00000000000000150000")
         self.assertEqual(floatformat(1.00000000000000015, 16), "1.0000000000000002")
 
+    def test_invalid_inputs(self):
+        cases = [
+            None,
+            "foo",  # random string
+            "error",  # random string with e in it
+            "¿Cómo esta usted?",  # non ascii
+            "1e",  # missing the exponent value after 'e'
+            "1e^2",  # uses '^' instead of a digit for the exponent
+            "1e2e3"  # multiple 'e' characters for exponentiation
+            "1.e2",  # misplaced decimal point
```

Comment:
There's a comma missing after `"1e2e3"`, once you add this in there is a failure for `"1.e2"` (which resolves to 100)

Reply:
Sorry about that, fixing while incorporating your next suggestion.